### PR TITLE
Implement player name

### LINKED ISSUE #1303: [FEATURE] Re implement player name
## 🚀 Feature
Related: https://github.com/lambdaclass/curse_of_mirra/pull/1040

- Until we have a centralized player database, each server will have its own.
- If the player tries to connect to a server and they have not created a user or haven't set a name before, it should prompt for name input before continuing to the MainScene.

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #1303 

## Motivation

Implement player name

Backend -> https://github.com/lambdaclass/mirra_backend/pull/369

## Summary of changes

Change player name popup
Delete unused scripts
Show playername ingame and in mainscreen.

## How has this been tested?

Describe the steps you performed to test this PR.

## Test additions / changes

List tests added/updated.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Comment by jazimp** [comment]
Bauti showed me how the design implementation looks, and it looks good! It has my visual approval 👍🏻

**Review by agustinesco — COMMENTED** [review]
The name is getting outside of the label in the main screen 
image

**Comment by agustinesco** [comment]
The game client players doesn't have their name
image

**Comment by agustinesco** [comment]
i think we should include the names in the prepare for battle screen too 
image

**Comment by agustinesco** [comment]
I think we should kind of adjust the text size on the kill feed to support longer names? 
image

**Review by agustinesco — CHANGES_REQUESTED** [review]
The killfeed is working as expected and the name is being displayer correctly! let's fix the previous comment and we're good to go!

i've tested this pr in both an android release and in the unity client

**Comment by BertovDev** [comment]
> The game client players doesn't have their name image

This was a think I forgot in the backend repo. All fixed!

**Review by agustinesco — APPROVED** [review]
LGTM! 

Not blocking but for long names the "you" tag is being overlap, this tag was temporal? 

image

File: client/Assets/Scripts/UI/EndGame/EndGameManager.cs [hunk 1]
```diff
@@ -97,7 +98,7 @@ void ShowMatchInfo()
         }
         else
         {
-            //defeaterPlayerName.text = GetDefeaterPlayerName();
+            // defeaterPlayerName.text = GetDefeaterPlayerName();
```
Comment (saugon) [inline client/Assets/Scripts/UI/EndGame/EndGameManager.cs:101]: Let's add this too  (it may need a reference visual from @jazimp)

Comment (jazimp) [reply client/Assets/Scripts/UI/EndGame/EndGameManager.cs:101]: refe name killed by
Font: Source Sans 3, variable: Regular 

Disclaimer: This is the quick solution to get it working now. In the future, a design will be proposed that includes all the match result information.

Comment (BertovDev) [reply client/Assets/Scripts/UI/EndGame/EndGameManager.cs]: Screenshot 2024-03-15 at 17 04 05

Comment (jazimp) [reply client/Assets/Scripts/UI/EndGame/EndGameManager.cs]: It has my visual approval 👍🏻

**Review by AngieDutra — COMMENTED** [review]
If we are adding the character name back to the game, aren't we supposed to add the button to change this name too? The game object for that is in title screen and the game object is called ChangeNameContainer

**Comment by BertovDev** [comment]
> If we are adding the character name back to the game, aren't we supposed to add the button to change this name too? The game object for that is in title screen and the game object is called ChangeNameContainer

I think it has a point but at the same time I would not continue adding complexity to this PR. We can move on and add it to another one. I like the suggestion indeed , what do you think to take advice from the team and maybe create an issue about it.

**Comment by BertovDev** [comment]
> Talked with our TL who suggested an issue is created for this game object to be an option within settings in main screen as soon as the PR adding that settings screen is in main. Do you mind creating that issue?

Suree

**Review by manucamejo — APPROVED** [review]

**Review by AngieDutra — APPROVED** [review]
Tested on Android and worked like a charm ✨

**Pull request merged**